File: app/src/main/java/brmobi/moop/features/publicacoes/ImageActivity.java
package brmobi.moop.features.publicacoes;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;

import com.squareup.picasso.Callback;
import com.squareup.picasso.NetworkPolicy;
import com.squareup.picasso.Picasso;

import brmobi.moop.model.rotas.RetrofitSingleton;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import brmobi.moop.R;
import uk.co.senab.photoview.PhotoViewAttacher;

public class ImageActivity extends AppCompatActivity {

    @BindView(R.id.imgFoto)
    ImageView imgFoto;


    private PhotoViewAttacher mAttacher;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_image);
        ButterKnife.bind(this);


        final String url = getIntent().getExtras().getString("bitmap");
        Picasso.with(this).load(RetrofitSingleton.BASE_URL + url).networkPolicy(NetworkPolicy.OFFLINE).into(imgFoto, new Callback() {
            @Override
            public void onSuccess() {

            }

            @Override
            public void onError() {
                Picasso.with(ImageActivity.this).load(url).into(imgFoto);
            }
        });
        mAttacher = new PhotoViewAttacher(imgFoto);
        mAttacher.setMaximumScale(3f);
        mAttacher.setOnViewTapListener(new PhotoViewAttacher.OnViewTapListener() {
            @Override
            public void onViewTap(View view, float x, float y) {
                ImageActivity.this.onBackPressed();
            }
        });
        mAttacher.setOnPhotoTapListener(new PhotoViewAttacher.OnPhotoTapListener() {
            @Override
            public void onPhotoTap(View view, float x, float y) {
                ImageActivity.this.onBackPressed();
            }
        });

    }

    @Override
    protected void onDestroy() {
        mAttacher.cleanup();
        super.onDestroy();
    }

    @OnClick(R.id.rootView)
    public void close(View view) {
        onBackPressed();
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }


}

File: app/src/main/java/brmobi/moop/ui/publication/ImageActivity.kt
package brmobi.moop.ui.publication

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import brmobi.moop.R
import brmobi.moop.utils.AppConstants
import com.squareup.picasso.Callback
import com.squareup.picasso.NetworkPolicy
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_image.*
import uk.co.senab.photoview.PhotoViewAttacher

class ImageActivity : AppCompatActivity() {


    private var mAttacher: PhotoViewAttacher? = null

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_image)
        val url = intent.extras!!.getString("bitmap")
        Picasso.with(this).load(AppConstants.BASE_URL + url!!).networkPolicy(NetworkPolicy.OFFLINE).into(imgFoto!!, object : Callback {
            override fun onSuccess() {

            }

            override fun onError() {
                Picasso.with(this@ImageActivity).load(url).into(imgFoto)
            }
        })
        mAttacher = PhotoViewAttacher(imgFoto)
        mAttacher!!.maximumScale = 3f
        mAttacher!!.onViewTapListener = PhotoViewAttacher.OnViewTapListener { view, x, y -> this@ImageActivity.onBackPressed() }
        mAttacher!!.onPhotoTapListener = PhotoViewAttacher.OnPhotoTapListener { view, x, y -> this@ImageActivity.onBackPressed() }
        rootView.setOnClickListener {
            onBackPressed()
        }

    }

    override fun onDestroy() {
        mAttacher!!.cleanup()
        super.onDestroy()
    }


}
